code/modules/reagents/chemistry/holder.dm

```diff
@@ -591,7 +591,8 @@
 							log_game("[key_name(C)] has started overdosing on [R.name] at [R.volume] units.")
 					var/is_addicted_to = addiction_list && is_type_in_list(R, addiction_list)
 					if(R.addiction_threshold)
-						if(R.volume >= R.addiction_threshold && !is_addicted_to)
+						var/addiction_prob = (R.volume / R.addiction_threshold) * 100
```

Comment:
I think this is probably not too terrible math wise but does make it much easer to abuse those low threshold addiction chems compared to high threshold chems. Stuff like nicotine, in the way you take it currently being entirely microdosages from cigarettes, has almost no chance of making you addicted. But meth, with it's 20u threshold you'll probably get addicted at 10 units.

The issue with addictions is that they're far, far too powerful when they occur to ever justify making them easier to induce into people, since that just becomes much easier to abuse as a weapon than as a tool. That's flipping the problem the other way, and these chems are already very powerful weapons in their own right. Worse, addiction can essentially render the body you are currently in a lost cause, since there are no medical methods to reverse addiction except time.

And most of the very powerful, weaponizable drugs have low addictions and effects that will absolutely kill you if you're addicted at all, with no way to stop it without someone sitting their ass in front of you feeding you drugs and doing constant brain surgery for about half a round.

Reply:
> medical methods to reverse addiction except time.

seems like the psychologist could have a part to play here doing rehab some how? anyway as it currently is the addiction threshold acts as what most chemists treat as the od in practice. maybe changing the effects around for a few chems would make it so players are willing to take the effects of an addiction for a longer buff on say meth.